### DISCUSSION TITLE
Text-based stamps with async_sign

### DISCUSSION THREAD

**Discussion by inui-sdt2** [MatthiasValvekens/pyHanko#88] [Q&A] [answered]
Dear @MatthiasValvekens 
Is there any way to generate signature appearance(Text-based stamps etc,) with async_sign_pdf?

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi @inui-sdt2,


The function `async_sign_pdf` in `pyhanko.sign.signers.functions` doesn't take any styling parameters, but that function is actually just a very thin wrapper around the `async_sign_pdf` method of the `pyhanko.sign.signers.pdf_signer.PdfSigner` class: https://github.com/MatthiasValvekens/pyHanko/blob/c5e2e6ff8252ae684fe42a98410fc62b1817ed31/pyhanko/sign/signers/functions.py#L129-L136.

If you instantiate your own `PdfSigner`, then you can pass in a `stamp_style` of your choosing. There's [an example in the documentation](https://pyhanko.readthedocs.io/en/latest/lib-guide/signing.html#text-based-stamps) showing how to do that---the example uses `sign_pdf` instead of `async_sign_pdf`, but as far as applying stamp styles goes, they're entirely analogous.
